app/controllers/puppetca_controller.rb

```diff
@@ -3,8 +3,7 @@ def index
     @proxy = find_proxy
     # expire cache if forced
     Rails.cache.delete("ca_#{@proxy.id}") if params[:expire_cache] == "true"
-    certs = find_certs
-    @certificates = certs.sort.paginate :page => params[:page], :per_page => Setting::General.entries_per_page
+    @certificates = find_certs
```

Comment:
Removed ```find_certs``` method and now returning all certs. Filtering is done only on the client side. Seems reasonable?